docs/pull-requests.md

```diff
@@ -0,0 +1,161 @@
+
+### Assume positive intent
+
+If you receive a comment that is brusque, has a negative tone, is or difficult to decipher, do your best to handle it gracefully and ask the reviewer for more information if necessary.
+
+### Point to updates
+
+As you are fulfilling others' requests, respond by linking to the commit that contains the changes you made. This gives reviewers the ability to check your work and allows the discussion to reach a resolution. (Tip: You can find commits you've pushed in the "Conversation" view of the pull request. If you copy a commit ID, then GitHub will link to it when you paste it in a comment. You can also just type the commit ID in your comment and GitHub will automatically link it.)
+
+```
+Good catch! Updated in abc1234.
+```
+
+### Employ alternate forms of communication
+
+If you sense that a conversation between you and the author is proceeding in an undesirable direction, consider reaching out to the person directly and offering to talk through your concerns on a video call.
+
+## On maintaining pull requests
+
+### Communicate takeovers
+
+If you need to take over and merge someone else's branch, let them know so they aren't surprised.
+
+### Rebase sparingly
+
+Once you've created a pull request, avoid amending, rebasing, or otherwise altering the history of the branch, but push up each new change as a new commit instead. This has two advantages:
+
+1. It keeps reviewers in the loop as changes are made. Keep in mind that they won't check your pull request as often as you will. If you've made a bunch of updates in response to feedback and now need a second round of reviews, then your reviewers can look over the list of most recent commits to catch up.
+2. It creates a smooth workflow with collaborators. If you're working with someone else but you change the history of the branch, they may have to now blow away their version of the branch and re-fetch it. This creates friction, causes frustration, and wastes time.
+
+Make sure that when you click the button on GitHub to update a branch with its base branch, you do not choose the "rebase" option.
+
+If you absolutely have to change the history of a pull request's branch, inform your reviewers and/or collaborators appropriately.
+
+## On merging pull requests
+
+### Clean up the commit message
+
+When you press the big green "Squash & Merge" button on GitHub, you will be presented with an option to modify the commit message before you merge.
+
+If you've taken the time to [write a good pull request description](#use-the-description-to-explain-the-need-and-solution), you may be surprised to find your description buried at the bottom of this message. This happens because of how GitHub performs the squash function: if you've pushed up any commits other than the initial commit, GitHub will generate a new commit message for you that simply repeats all of their names. **As is, this unordered list will be persisted in the repository for all time.**
```

Comment:
Nit: We may want to qualify this somewhat, since the default Squash & Merge message is now configurable on a per-repository basis. There are a couple of different options now. Though we don't want to get too into the weeds on that, this is still good advice despite the specifics.

Reply:
Addressed here: https://github.com/MetaMask/contributor-docs/pull/46